ledger/ledger-api-client/src/it/scala/com/digitalasset/ledger/client/UpgradesIT.scala

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.client
+
+import com.daml.integrationtest.CantonFixture
+import org.scalatest.Inside
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+import scala.concurrent.{Future}
+import com.google.protobuf.ByteString
+import com.daml.bazeltools.BazelRunfiles._
+import java.io.File
+import java.io.FileInputStream
+import org.scalatest.Suite
+import org.scalatest.compatible.Assertion
+
+import scala.io.Source
+import com.daml.lf.data.Ref.PackageId
+
+import com.daml.lf.archive.{DarReader}
+import scala.util.{Success, Failure}
+
+final class UpgradesIT extends AsyncWordSpec with Matchers with Inside with CantonFixture {
+  self: Suite =>
+  override lazy val devMode = true;
+  override val cantonFixtureDebugMode = true;
```

Comment:
I need to canton logs currently for error message checking - blocked on https://github.com/digital-asset/daml/pull/18102/files#r1466503011

Reply:
I don't think this flag stops the logs from existing, it just doesn't delete them

Reply:
I think it stops the logs from existing because `debugOptions` in `CantonRunner` depends on `debug`, which is `cantonFixtureDebugMode` in `CantonFixture`. I've added a new flag which forces temporarily file removal even when debug is set.

fixed

Reply:
Oh i see, fair enough - nice :)